### DISCUSSION TITLE
How to save script on Boost hub?

### DISCUSSION THREAD

**Discussion by vjchan99** [pybricks/support#1116] [Q&A]
Hi, currently, I need to run my code from pybricks to my boost hub. How do I add the functionality to store the program on the hub so that I can use it autonomously?

Thanks.

**Comment by dlech** [comment]
Once you download and run a program and then stop it, you can press the button on the hub to start the program again.

When you power off the hub by pressing and holding the green button the program is saved to non-volatile memory and will still be there when you turn the hub on again.

**Reply by vjchan99** [reply]
Thanks. Do I need to add code to add that functionality? Right now, if I close the pybricks app and then turn the robot on, it seems to be looking for a bluetooth connection.

**Reply by dlech** [reply]
No, you don't need to do anything extra in the program. When it seems to be looking for a Bluetooth connection, just press the green button to run the program.

**Reply by vjchan99** [reply]
Thanks so much! I didn't realise I needed to press the green button again. Works now.
